plugin/pkg/admission/serviceaccount/admission.go

```diff
@@ -198,6 +198,9 @@ func (s *Plugin) Validate(ctx context.Context, a admission.Attributes, o admissi
 					if projSource.ServiceAccountToken != nil {
 						return admission.NewForbidden(a, fmt.Errorf("a mirror pod may not use ServiceAccountToken volume projections"))
 					}
+					if projSource.ClusterTrustBundle != nil {
+						return admission.NewForbidden(a, fmt.Errorf("a mirror pod may not use ClusterTrustBundle volume projections"))
+					}
```

Comment:
xref https://github.com/kubernetes/kubernetes/issues/103587 and example at https://github.com/liggitt/kubernetes/commits/kubelet-static-api-reference

Reply:
also guard against this in node restriction admission:

https://github.com/kubernetes/kubernetes/blob/0270fc75d0f59e1f5a31a810737ac35ac3eeb8a3/plugin/pkg/admission/noderestriction/admission.go#L247-L265

Reply:
I've blocked references to ClusterTrustBundles in the noderestriction admission plugin.  

It looks like the change for Kubelet to reject these references in static pods earlier hasn't landed yet.

Reply:
> I've blocked references to ClusterTrustBundles in the noderestriction admission plugin.

that looks good

> It looks like the change for Kubelet to reject these references in static pods earlier hasn't landed yet.

At the very least, we should avoid making the issue worse in this PR by adding a check to tryDecodeSinglePod that rejects static pods that include clusterTrustBundle volumes